### PR TITLE
added cauchy link function

### PR DESCRIPTION
**Pull request by StrikerRUS**
Refer to #228.

### CODE REVIEW

File: m2cgen/assemblers/fallback_expressions.py [hunk 1]
```diff
@@ -66,6 +64,118 @@ def log1p(expr):
         utils.div(utils.mul(expr, ast.LogExpr(expr1p)), expr1pm1))
 
 
+def atan(expr):
```
Comment (izeigerman) [inline m2cgen/assemblers/fallback_expressions.py:67]: This looks super scary :D

**Review by izeigerman — APPROVED** [review]
This looks great, thanks!

File: m2cgen/assemblers/fallback_expressions.py (hunk 1 above)
Comment (StrikerRUS) [reply m2cgen/assemblers/fallback_expressions.py:67]: Yeah, I know! 😃 

But it was the only one I could find with appropriate accuracy. Other either have error about 1e-4 - 1e-6 or look even more scary. I even cannot remember how many trials I've made. There were Chebyshev, CORDIC, Taylor, Remez (in [Sollya package](http://sollya.gforge.inria.fr/sollya-2.9/help.php?name=remez))...

Comment (izeigerman) [reply m2cgen/assemblers/fallback_expressions.py:67]: Wow, impressive. Amazing work as always 👍

Comment (StrikerRUS) [reply m2cgen/assemblers/fallback_expressions.py:67]: Thank you!

**Pull request merged**